Assets/Features/Building/Scripts/Grid/Grid.cs

```diff
@@ -0,0 +1,391 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Features.Building.Scripts.Datatypes;
+using Features.Managers;
+using Implementation.Workers.TaskCommands;
+using UnityEngine;
+using UnityEngine.Tilemaps;
+using TileData = Features.Building.Scripts.Datatypes.TileData;
+
+namespace Features.Building.Scripts.Grid
+{
+    public class Grid : MonoBehaviour
+    {
+        /// <summary>
+        /// contains the whole grid. the 3d array is used as the following [x,y,z]
+        /// x = x-axis
+        /// y = y-axis
+        /// z = z-axis for multiple layers for example floor tile layer and decoration layer
+        /// </summary>
+        private CellData[,,] _cells;
+
+        [SerializeField] private BuildableAtlas _atlas;
+        [SerializeField] private Tilemap _tilemap;
+
+        [SerializeField] private Vector3Int _gridSize;
+        [SerializeField] private float _cellSize;
+
+        [SerializeField] private List<List<Vector3Int>> _cellGroup;
+
+        /// <summary>
+        /// If true the map will be updated at the end of the frame and set to false
+        /// </summary>
+        private bool _mapUpdated;
+
+
+        public Vector3Int GridSize => _gridSize;
+        public float CellSize => _cellSize;
+
+        void Start()
+        {
+            if (_atlas == null)
+                Debug.LogError("No atlas is assigned!");
+
+            _cells = new CellData[_gridSize.x, _gridSize.y, _gridSize.z];
+            _cellGroup = new List<List<Vector3Int>>();
+            PopulateCells();
+        }
+
+        /// <summary>
+        /// Creates a flattend 2d array to see if it is traversable
+        /// </summary>
+        /// <returns>A flattend 2d bool array with false as traversable</returns>
+        public bool[,] UnTraversable()
```

Comment:
A name Like `GetUntraversable()` might be more suitable, although it might be more straightforward to return a 2d array containing traversable tiles instead of non traversable tiles. At least, when reading it